lib/src/settings.rs

```diff
@@ -26,13 +26,38 @@ pub struct UserSettings {
     config: config::Config,
     timestamp: Option<Timestamp>,
     rng: Arc<JJRng>,
+    git_settings: GitSettings,
 }
 
 #[derive(Debug, Clone)]
 pub struct RepoSettings {
     _config: config::Config,
 }
 
+#[derive(Debug, Clone)]
+pub struct GitSettings {
+    pub auto_local_branch: bool,
+}
+
+impl GitSettings {
+    pub fn from_config(config: &config::Config) -> Self {
+        let mut settings = GitSettings::default();
+
+        if let Ok(auto_local_branch) = config.get_bool("git.auto-local-branch") {
+            settings.auto_local_branch = auto_local_branch;
+        }
```

Comment:
Remember that old conversation we had about public/private checklists? I was ticking these off as I resolved them privately. Looks like I shouldn't do that any more because it's confusing haha.